### ISSUE TITLE
Keras model from python successfully imported but getting input shape error.

### ISSUE THREAD

**Issue by vishavdhanjal** [deeplearning4j/deeplearning4j#6274]
Input shape of model is "batch_input_shape": [null, 300, 1].

But couldn't make input shape like this. Kindly share some helpful material for such input shape (for random number). 

OTHER DETAILS:
Rank: 2,Offset: 0
 Order: c Shape: [300,1],  stride: [1,1]
Invalid size: cannot get size of dimension 2 for rank 2 NDArray (array shape: [300, 1])

**Comment by maxpumperla** [comment]
how about you provide me with an actual model? what's your error message? what dl4j version etc.

**Comment by maxpumperla** [comment]
in any case you need a minibatch dim. so [mb, 300, 1] for instance

**Comment by vishavdhanjal** [comment]
ACTUAL CODE IN PYTHON
model = Sequential()
model.add(Conv1D(16,kernel_size=3, input_shape=(300,1)))
model.add(Conv1D(32,kernel_size=3))
model.add(Flatten())
model.add(Dense(2000, activation='relu'))#, input_dim=300))
model.add(Dense(1000, activation='relu'))
model.add(Dense(500, activation='relu'))
model.add(Dense(250, activation='relu'))
model.add(Dense(5, activation='sigmoid'))
model.summary()
model.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])

model.fit(trainX, trainY, batch_size = 10, epochs = 10, verbose=1,
          validation_data=(testX, testY))

MODEL SUMMARY IN PYTHON
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
conv1d_3 (Conv1D)            (None, 298, 16)           64        
_________________________________________________________________
conv1d_4 (Conv1D)            (None, 296, 32)           1568      
_________________________________________________________________
flatten_2 (Flatten)          (None, 9472)              0         
_________________________________________________________________
dense_6 (Dense)              (None, 2000)              18946000  
_________________________________________________________________
dense_7 (Dense)              (None, 1000)              2001000   
_________________________________________________________________
dense_8 (Dense)              (None, 500)               500500    
_________________________________________________________________
dense_9 (Dense)              (None, 250)               125250    
_________________________________________________________________
dense_10 (Dense)             (None, 5)                 1255      
=================================================================
Total params: 21,575,637
Trainable params: 21,575,637
Non-trainable params: 0

MODEL SUMMARY IN JAVA

==========================================================================================================================================================================================================================================================
LayerName (LayerType)                   nIn,nOut  TotalParams ParamsShape                             
==========================================================================================================================================================================================================================================================
conv1d_3 (Convolution1DLayer)           1,16        64        b:{1,16}, W:{16,1,3,1}                  
conv1d_4 (Convolution1DLayer)           16,32       1568      b:{1,32}, W:{32,16,3,1}                 
dense_6 (DenseLayer)                    32,2000     66000     W:{32,2000}, b:{1,2000}                 
dense_7 (DenseLayer)                    2000,1000   2001000   W:{2000,1000}, b:{1,1000}               
dense_8 (DenseLayer)                    1000,500    500500    W:{1000,500}, b:{1,500}                 
dense_9 (DenseLayer)                    500,250     125250    W:{500,250}, b:{1,250}                  
dense_10 (DenseLayer)                   250,5       1255      W:{250,5}, b:{1,5}                      
dense_10_loss (LossLayer)               -,-         0         -                                       
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
            Total Parameters:  2695637
        Trainable Parameters:  2695637
           Frozen Parameters:  0

**Comment by vishavdhanjal** [comment]
I am taking mb=1 in my case. But it is not working.

**Comment by maxpumperla** [comment]
just curious (have to check later), does [1,1,1,300] as input shape work?

**Comment by vishavdhanjal** [comment]
It worked. Actually I was confused in [minibatch,inputDepth,height,width] and was making combination of (1,1,1,300). Thanks it was a great help :). [I am having some issue in prediction but i will resolve that one myself. Thanks once again]

**Comment by maxpumperla** [comment]
cool. I'm closing this then.
